File: shared-atlas-dart/lib/src/interfaces/web_socket_error_code.dart
enum WebSocketErrorCode {
  // @JsonValue('4010')
  AUTH_TOKEN_MISSING('4010'),
  // @JsonValue('4011')
  AUTH_TOKEN_INVALID('4011'),
  // @JsonValue('4012')
  AUTH_TOKEN_EXPIRED('4012'),
  // @JsonValue('4030')
  PERMISSION_DENIED('4030'),
  // @JsonValue('4031')
  RATE_LIMIT_EXCEEDED('4031'),
  // @JsonValue('4032')
  CONCURRENT_LIMIT_REACHED('4032'),
  // @JsonValue('4220')
  INVALID_PAYLOAD('4220'),
  // @JsonValue('4221')
  NOTE_NOT_FOUND('4221'),
  // @JsonValue('4222')
  SESSION_NOT_FOUND('4222'),
  // @JsonValue('5000')
  INTERNAL_ERROR('5000'),
  // @JsonValue('5001')
  DATABASE_ERROR('5001'),
  // @JsonValue('5002')
  YJS_SYNC_FAILED('5002');

  final String value;
  const WebSocketErrorCode(this.value);

  factory WebSocketErrorCode.fromJson(dynamic json) {
    return WebSocketErrorCode.values.firstWhere((e) => e.value == json.toString(), orElse: () => WebSocketErrorCode.values.first);
  }

  String toJson() => value;
}

File: shared-atlas-rust/src/interfaces/web_socket_error_code.rs
use serde::{Serialize, Deserialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum WebSocketErrorCode {
    #[serde(rename = "4010")]
    AuthTokenMissing,
    #[serde(rename = "4011")]
    AuthTokenInvalid,
    #[serde(rename = "4012")]
    AuthTokenExpired,
    #[serde(rename = "4030")]
    PermissionDenied,
    #[serde(rename = "4031")]
    RateLimitExceeded,
    #[serde(rename = "4032")]
    ConcurrentLimitReached,
    #[serde(rename = "4220")]
    InvalidPayload,
    #[serde(rename = "4221")]
    NoteNotFound,
    #[serde(rename = "4222")]
    SessionNotFound,
    #[serde(rename = "5000")]
    InternalError,
    #[serde(rename = "5001")]
    DatabaseError,
    #[serde(rename = "5002")]
    YjsSyncFailed,
}